File: src/main/java/pt/ua/nicevolunteers/volunteer/service/PointsService.java
package pt.ua.nicevolunteers.volunteer.service;

import java.util.List;
import java.util.UUID;

import org.springframework.stereotype.Service;

import pt.ua.nicevolunteers.volunteer.domain.points.PointsTransaction;
import pt.ua.nicevolunteers.volunteer.repository.PointsTransactionRepository;

@Service
public class PointsService {

    private final PointsTransactionRepository repository;

    public PointsService(PointsTransactionRepository repository) {
        this.repository = repository;
    }

    public int getCurrentBalance(UUID volunteerId) {
        return repository.findByVolunteerIdOrderByTimestampAsc(volunteerId)
                .stream()
                .mapToInt(PointsTransaction::getPoints)
                .sum();
    }

    public List<PointsTransaction> getHistory(UUID volunteerId) {
        return repository.findByVolunteerIdOrderByTimestampAsc(volunteerId);
    }

    public List<PointsTransaction> getHistoryForVolunteer(UUID requesterId, UUID targetVolunteerId) {
        if (!requesterId.equals(targetVolunteerId)) {
            throw new SecurityException("Cannot access another volunteer's points history");
        }
        return getHistory(targetVolunteerId);
    }
}

File: frontend/src/services/pointsService.js
import api from "../api/api";

const pointsService = {
  /**
   * Get points balance for a volunteer
   * @param {string} volunteerId - UUID of the volunteer
   * @returns {Promise<Object>} Points balance response with {volunteerId, balance}
   */
  async getBalance(volunteerId) {
    const response = await api.get(`/api/points/balance/${volunteerId}`);
    return response.data;
  },

  /**
   * Get points transaction history for a volunteer
   * @param {string} volunteerId - UUID of the volunteer
   * @returns {Promise<Array>} List of points transactions
   */
  async getHistory(volunteerId) {
    const response = await api.get(`/api/points/history/${volunteerId}`);
    return response.data;
  }
};

export default pointsService;
